### ISSUE TITLE
OS X 10.9.5 error: invalid value 'c++14' in '-std=c++14'

### ISSUE THREAD

**Issue by superclarkk** [BartVandewoestyne/oaconvert#135]
Following on from #134, I tried to install oaconvert on my mabook running OS X 10.9.5. After installing clang using macports, I had the same error, `/bin/sh: /usr/bin/clang++-3.5: No such file or directory`. I tried to fudge it, by symlinking from clang to clang++-3.5, but this resulted in `error: invalid value 'c++14' in '-std=c++14'`.

```
macbookpro:src root# port install clang-3.5
--->  Computing dependencies for clang-3.5
--->  Dependencies to be installed: clang_select ld64 ld64-latest libcxx llvm-3.5 llvm_select
--->  Fetching archive for clang_select
--->  Attempting to fetch clang_select-1.0_0.darwin_13.noarch.tbz2 from http://lil.fr.packages.macports.org/clang_select
--->  Attempting to fetch clang_select-1.0_0.darwin_13.noarch.tbz2.rmd160 from http://lil.fr.packages.macports.org/clang_select
--->  Installing clang_select @1.0_0
--->  Activating clang_select @1.0_0
--->  Cleaning clang_select
--->  Fetching archive for libcxx
--->  Attempting to fetch libcxx-3.6.0_0.darwin_13.x86_64.tbz2 from http://lil.fr.packages.macports.org/libcxx
--->  Attempting to fetch libcxx-3.6.0_0.darwin_13.x86_64.tbz2.rmd160 from http://lil.fr.packages.macports.org/libcxx
--->  Installing libcxx @3.6.0_0
--->  Activating libcxx @3.6.0_0
--->  Cleaning libcxx
--->  Fetching archive for llvm_select
--->  Attempting to fetch llvm_select-1.0_0.darwin_13.noarch.tbz2 from http://lil.fr.packages.macports.org/llvm_select
--->  Attempting to fetch llvm_select-1.0_0.darwin_13.noarch.tbz2.rmd160 from http://lil.fr.packages.macports.org/llvm_select
--->  Installing llvm_select @1.0_0
--->  Activating llvm_select @1.0_0
--->  Cleaning llvm_select
--->  Fetching archive for llvm-3.5
--->  Attempting to fetch llvm-3.5-3.5.2_4.darwin_13.x86_64.tbz2 from http://lil.fr.packages.macports.org/llvm-3.5
--->  Attempting to fetch llvm-3.5-3.5.2_4.darwin_13.x86_64.tbz2.rmd160 from http://lil.fr.packages.macports.org/llvm-3.5
--->  Installing llvm-3.5 @3.5.2_4
--->  Activating llvm-3.5 @3.5.2_4
--->  Cleaning llvm-3.5
--->  Fetching archive for ld64-latest
--->  Attempting to fetch ld64-latest-242_0+llvm35.darwin_13.x86_64.tbz2 from http://lil.fr.packages.macports.org/ld64-latest
--->  Attempting to fetch ld64-latest-242_0+llvm35.darwin_13.x86_64.tbz2.rmd160 from http://lil.fr.packages.macports.org/ld64-latest
--->  Installing ld64-latest @242_0+llvm35
--->  Activating ld64-latest @242_0+llvm35
--->  Cleaning ld64-latest
--->  Fetching archive for ld64
--->  Attempting to fetch ld64-2_0.darwin_13.x86_64.tbz2 from http://lil.fr.packages.macports.org/ld64
--->  Attempting to fetch ld64-2_0.darwin_13.x86_64.tbz2.rmd160 from http://lil.fr.packages.macports.org/ld64
--->  Installing ld64 @2_0
--->  Activating ld64 @2_0
--->  Cleaning ld64
--->  Fetching archive for clang-3.5
--->  Attempting to fetch clang-3.5-3.5.2_5+analyzer+arm_runtime.darwin_13.x86_64.tbz2 from http://lil.fr.packages.macports.org/clang-3.5
--->  Attempting to fetch clang-3.5-3.5.2_5+analyzer+arm_runtime.darwin_13.x86_64.tbz2.rmd160 from http://lil.fr.packages.macports.org/clang-3.5
--->  Installing clang-3.5 @3.5.2_5+analyzer+arm_runtime
--->  Activating clang-3.5 @3.5.2_5+analyzer+arm_runtime
--->  Cleaning clang-3.5
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  No broken files found.
macbookpro:src root# port search clang
Bear @2.0.3 (devel)
    Bear is a tool to generate compilation database for clang tooling.

clang-2.9 @2.9_15 (lang)
    this port is only a stub and has been made obsolete by clang-3.4

clang-3.0 @3.0_14 (lang)
    this port is only a stub and has been made obsolete by clang-3.4

clang-3.1 @3.1_9 (lang)
    this port is only a stub and has been made obsolete by clang-3.4

clang-3.2 @3.2_4 (lang)
    this port is only a stub and has been made obsolete by clang-3.4

clang-3.3 @3.3_8 (lang)
    C, C++, Objective C and Objective C++ compiler

clang-3.4 @3.4.2_8 (lang)
    C, C++, Objective C and Objective C++ compiler

clang-3.5 @3.5.2_5 (lang)
    C, C++, Objective C and Objective C++ compiler

clang-3.6 @3.6.1_1 (lang)
    C, C++, Objective C and Objective C++ compiler

clang-3.7 @3.7-r241984 (lang)
    C, C++, Objective C and Objective C++ compiler

clang_select @1.0 (sysutils)
    common files for selecting default clang version

mpich-clang @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-clang33 @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-clang34 @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-clang35 @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-clang36 @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-clang37 @3.1.4_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang33 @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang34 @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang35 @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang36 @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

mpich-devel-clang37 @3.2b3_1 (science, parallel, net)
    Message Passing Interface (MPI) Library

openmpi-clang @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-clang33 @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-clang34 @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-clang35 @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-clang36 @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-clang37 @1.7.5_5 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang33 @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang34 @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang35 @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang36 @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

openmpi-devel-clang37 @1.9a1_34660 (science, parallel, net)
    A High Performance Message Passing Library

Found 35 ports.

macbookpro:bin root# cd oaconvert/
macbookpro:oaconvert root# rm CMakeCache.txt 
macbookpro:oaconvert root# cmake .
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   airspace

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /usr/local/bin/oaconvert
macbookpro:oaconvert root# cd src/
macbookpro:src root# make
[  5%] Building CXX object src/CMakeFiles/airspace.dir/Airspace.cpp.o
/bin/sh: /usr/bin/clang++-3.5: No such file or directory
make[2]: *** [src/CMakeFiles/airspace.dir/Airspace.cpp.o] Error 127
make[1]: *** [src/CMakeFiles/airspace.dir/all] Error 2
make: *** [all] Error 2
macbookpro:src root# ls -la /usr/bin |grep -i clang
lrwxr-xr-x     1 root   wheel         7  6 Jun  2014 c++ -> clang++
lrwxr-xr-x     1 root   wheel         5  6 Jun  2014 cc -> clang
-rwxr-xr-x     1 root   wheel     14224  6 Jun  2014 clang
-rwxr-xr-x     1 root   wheel     14240  6 Jun  2014 clang++
lrwxr-xr-x     1 root   wheel         7  6 Jun  2014 llvm-g++ -> clang++
lrwxr-xr-x     1 root   wheel         5  6 Jun  2014 llvm-gcc -> clang
macbookpro:src root# ln -s /usr/bin/clang /usr/bin/clang-3.5
macbookpro:src root# ln -s /usr/bin/clang++ /usr/bin/clang++-3.5
macbookpro:src root# pwd
/usr/local/bin/oaconvert/src
macbookpro:src root# make
[  5%] Building CXX object src/CMakeFiles/airspace.dir/Airspace.cpp.o
error: invalid value 'c++14' in '-std=c++14'
make[2]: *** [src/CMakeFiles/airspace.dir/Airspace.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/airspace.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by grazhopper** [comment]
It is in general not that good an idea for a cmake managed project to force the compiler (ok, sometimes it is, but in general, if the idea is to compile a certain project on many different platforms and systems, it is not). If you simply want to get oaconvert to compile, edit the CMakeLists.txt file and remove the line in which  CMAKE_CXX_COMPILER is set and the lines with CMAKE_CXX_FLAGS/CMAKE_CXX_FLAGS_DEBUG. Of course, this is not the "clean" solution, but it will get you somewhere.

**Comment by BartVandewoestyne** [comment]
Fixed by commit #139 by grazhopper.
